x/erc20/keeper/msg_server.go

```diff
@@ -21,48 +21,6 @@ import (
 
 var _ types.MsgServer = &Keeper{}
 
-// ConvertCoin converts native Cosmos coins into ERC20 tokens for both
-// Cosmos-native and ERC20 TokenPair Owners
-func (k Keeper) ConvertCoin(
-	goCtx context.Context,
-	msg *types.MsgConvertCoin,
-) (*types.MsgConvertCoinResponse, error) {
-	ctx := sdk.UnwrapSDKContext(goCtx)
-
-	// Error checked during msg validation
-	receiver := common.HexToAddress(msg.Receiver)
-	sender := sdk.MustAccAddressFromBech32(msg.Sender)
-
-	pair, err := k.MintingEnabled(ctx, sender, receiver.Bytes(), msg.Coin.Denom)
-	if err != nil {
-		return nil, err
-	}
-
-	// Remove token pair if contract is suicided
-	erc20 := common.HexToAddress(pair.Erc20Address)
-	acc := k.evmKeeper.GetAccountWithoutBalance(ctx, erc20)
-
-	if acc == nil || !acc.IsContract() {
-		k.DeleteTokenPair(ctx, pair)
-		k.Logger(ctx).Debug(
-			"deleting selfdestructed token pair from state",
-			"contract", pair.Erc20Address,
-		)
-		// NOTE: return nil error to persist the changes from the deletion
-		return nil, nil
-	}
-
-	// Check ownership and execute conversion
-	switch {
-	case pair.IsNativeCoin():
-		return k.convertCoinNativeCoin(ctx, pair, msg, receiver, sender) // case 1.1
-	case pair.IsNativeERC20():
-		return k.convertCoinNativeERC20(ctx, pair, msg, receiver, sender) // case 2.2
```

Comment:
this logic is still necessary for EVM ERC-20s

Reply:
Was going to port this into the new IBC callback on a separate PR